core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -331,6 +333,9 @@ public GetResult get(Get get, Function<String, Searcher> searcherFactory) throws
         try (ReleasableLock lock = readLock.acquire()) {
             ensureOpen();
             if (get.realtime()) {
+                if (isRealtimeGetEnabled == false) {
+                    throw new IllegalArgumentException("get with realtime=true is disabled on this index");
```

Comment:
Should we just default to false if realtime gets are disabled? I know we're removing all the crazy conditional defaults but this one doesn't seem so crazy because the user configures the index.